scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -77,24 +71,77 @@ const createDeleteSuggestion = (state, {data}) => {
     const selectionLength = selection.getEndOffset() - selection.getStartOffset();
 
     if (selectionLength === 0) {
-        editorState = setDeleteSuggestionForCharacter(editorState, data);
-    } else {
+        editorState = changeEditorSelection(editorState, -1, 0, false);
+    }
+    editorState = deleteCurrentSelection(editorState, data);
+
+    return onChange(state, editorState, true);
+};
+
+/**
+ * @ngdoc method
+ * @name pasteAddSuggestion
+ * @param {Object} state
+ * @param {Object} content - the suggestion added content
+ * @param {Object} data - info about the author of suggestion
+ * @return {Object} returns new state
+ * @description Add a new suggestion of type ADD based on content.
+ */
+const pasteAddSuggestion = (state, {content, data}) => {
+    const acceptedStyles = ['BOLD', 'ITALIC', 'UNDERLINE'];
+    let {editorState} = state;
+    const blockMap = content.getBlockMap();
+
+    editorState = deleteCurrentSelection(editorState, data);
+
+    blockMap.forEach((block) => {
```

Comment:
I don't understand how this works - it seems like it removes the selected characters - ok - but then it set add suggestion for all non atomic blocks and every character there?

Reply:
Yes, the suggestion entity key is set for every character.

Reply:
but how is this limited to only content which was pasted?

Reply:
Html is converted in an editor content structure that is received here(content parameter).

Reply:
so content is a contentState with pasted content only?

Reply:
yes